tests/JITBenchmark.java

```diff
@@ -40,22 +44,125 @@ public static void writeByteArrayOutputStream() {
     System.out.println("Length: " + stream.size());
   }
 
+  public static void concatStrings() {
+    String s = "";
+    int count = size / 16;
+    for (int i = 0; i < count; i++) {
+      s += "X";
+    }
+    System.out.println("Length: " + s.length());
+  }
+
+  public static void getBytes() {
+    String s = "getBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytes";
+    int count = size / 4;
+    byte [] bytes = null;
+    for (int i = 0; i < count; i++) {
+      bytes = s.getBytes();
+    }
+    System.out.println("Length: " + bytes.length);
+  }
+
+  public static void synch() {
+    int count = size * 10;
+    byte [] bytes = null;
+    Object o = new Object();
+    int sum = 0;
+    for (int i = 0; i < count; i++) {
+      synchronized (o) {
+        sum++;
+      }
+      synchronized (o) {
+        sum++;
+      }
+      synchronized (o) {
+        sum++;
+      }
+      synchronized (o) {
+        sum++;
+      }
+      synchronized (o) {
+        sum++;
+      }
+      synchronized (o) {
+        sum++;
+      }
+    }
+    System.out.println("Sum: " + sum);
+  }
+
+  public static void hashtable() {
+    int count = size * 2;
+    Hashtable hash = new Hashtable();
+    Object o = new Object();
+    String [] names = {
+      "hello", "world", "hello1", "world2", "hello3", "world4", "hello5", "world6"
+    };
+    for (int i = 0; i < count; i++) {
+      String name = names[i % names.length];
+      hash.put(name, o);
+      hash.get(name);
+      hash.put(name, o);
+      hash.get(name);
+      hash.put(name, o);
+      hash.get(name);
+      hash.put(name, o);
+      hash.get(name);
+    }
+  }
+
+  public static void arrayTypeCheck() {
+    A [] array = new A [1024];
+    A a = new A();
+    B b = new B();
+    int count = size * 10;
+    for (int i = 0; i < count - 1; i++) {
+      array[i % 1024] = a;
+      array[i % 1024] = b;
+    }
+  }
+
   public static void main(String[] args) {
     long start = 0;
 
     start = System.currentTimeMillis();
```

Comment:
In benchmarks, we should use JVM.monotonicTimeMillis (it uses performance.now() instead of Date.now())